server_addon/houdini/client/ayon_houdini/api/plugin.py

```diff
@@ -349,3 +354,21 @@ def apply_settings(self, project_settings):
 
         for key, value in settings.items():
             setattr(self, key, value)
+
+
+class HoudiniLoader(load.LoaderPlugin):
+    """Base class for most of the Houdini load plugins."""
+
+    settings_category = SETTINGS_CATEGORY
+
+
+class HoudiniInstancePlugin(pyblish.api.InstancePlugin):
+    """Base class for most of the Houdini instance publish plugins."""
+
+    settings_category = SETTINGS_CATEGORY
+
+
+class HoudiniContextPlugin(pyblish.api.ContextPlugin):
+    """Base class for most of the Houdini context publish plugins."""
+
+    settings_category = SETTINGS_CATEGORY
```

Comment:
Admittedly the extra inheritance here to me isn't really warranted. If it were more than one attribute, and more custom logic I'd understand. In this case I'd avoid from refraining for more inheriting the custom plug-in and instead just add `settings_category = "houdini"` to each OR maybe set it using the `SETTINGS_CATEGORY` constant.

This hardly feels like anything that _should_ change for the Houdini integration over time and if it does, then batch changing the attributes is just as trivial code-wise.

Reply:
Both approaches have caviads. We've discussed it with @antirotor and we don't have strong opinion about it. Both options are viable.

During the discussion we've came to semi-conclusion that maybe having base classes in hosts that have a lot of plugins might be better as you don't have to think about adding `settings_category` when creating new plugin, just use the base plugin class.

Reply:
#### Add hosts?

If we're going for base classes here. Should we also add `hosts = ["houdini"]` for those for which it is relevant?

#### Docstrings

Also, the docstrings state "for most of the Houdini plugins" - when shouldn't it be used? Should we remove the "most of the" part of the docstrings and just make it for example """Base class for Houdini context publish plugins."""

Reply:
Let's fix up the docstrings as stated and add the `hosts = ["houdini"]` as well, similar [as done in the Maya](https://github.com/ynput/ayon-core/pull/546/files#diff-4144b58e6210be392766cdb86c00e3e3f9a4fadac21fd2ac44267103d9130c83R1030-R1054)

Reply:
What about the Extractors ? 
Should I add a base class for extractors to be similar to Maya ?  Knowing that `publish.Extractor` is only an instance plugin with an importable function. 

https://github.com/ynput/ayon-core/blob/0594f49f4f03b2127db6224b8792bab078ff326e/client/ayon_core/pipeline/publish/publish_plugins.py#L271-L290